lib/node_modules/@stdlib/math/base/special/asinf/lib/main.js

```diff
@@ -0,0 +1,144 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The original C code, long comment, copyright, license, and constants are from [Cephes]{@link http://www.netlib.org/cephes}. The implementation follows the original, but has been modified for JavaScript.
+*
+* ```text
+* Copyright 1984, 1995, 2000 by Stephen L. Moshier
+*
+* Some software in this archive may be from the book _Methods and Programs for Mathematical Functions_ (Prentice-Hall or Simon & Schuster International, 1989) or from the Cephes Mathematical Library, a commercial product. In either event, it is copyrighted by the author. What you see here may be used freely but it comes with no support or guarantee.
+*
+* Stephen L. Moshier
+* moshier@na-net.ornl.gov
+* ```
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnanf = require( '@stdlib/math/base/assert/is-nanf' );
+var sqrtf = require( '@stdlib/math/base/special/sqrtf' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var PI02F = require( '@stdlib/constants/float32/half-pi' );
+var polyp = require( './poly_p.js' );
+
+
+// MAIN //
+
+/**
+* Computes the arcsine of a single-precision floating-point number.
+*
+* ## Method
+*
+* -   A rational function of the form
+*
+*     ```tex
+*     x + x^3 \frac{P(x^2)}{Q(x^2)}
+*     ```
+*
+*     is used for \\(\|x\|\\) in the interval \\(\[0, 0.5\]\\). If \\(\|x\| > 0.5\\), it is transformed by the identity
+*
+*     ```tex
+*     \operatorname{asinf}(x) = \frac{\pi}{2} - 2 \operatorname{asinf}( \sqrt{ (1-x)/2 } )
+*     ```
+*
+* ## Notes
+*
+* -   Relative error:
+*
+*     | arithmetic | domain | # trials | peak    | rms     |
+*     |:-----------|:-------|:---------|:--------|:--------|
+*     | IEEE       | -1, 1  | 10^5     | 1.4e-7  | 4.2e-8  |
+*
+* @param {number} x - input value
+* @returns {number} arcsine (in radians)
+*
+* @example
+* var v = asinf( 0.0 );
+* // returns ~0.0
+*
+* @example
+* var v = asinf( 3.14/4.0 );
+* // returns ~0.903
+*
+* @example
+* var v = asinf( -3.14/6.0 );
+* // returns ~-0.551
+*
+* @example
+* var v = asinf( NaN );
+* // returns NaN
+*/
+function asinf( x ) {
+	var flag;
+	var sgn;
+	var a;
+	var z;
+
+	if ( isnanf( x ) ) {
+		return NaN;
+	}
+	x = float64ToFloat32( x );
+	if ( x > 0.0 ) {
+		sgn = 1;
+		a = x;
+	}
+	else {
+		sgn = -1;
+		a = -x;
+	}
+	if ( a > 1.0 ) {
+		return 0.0;
+	}
+	if ( a < 1.0e-4 ) {
+		z = float64ToFloat32( a );
+		if ( sgn < 0 ) {
+			z = -z;
+		}
+		return z;
+	}
+	if ( a > 0.5 ) {
+		z = float64ToFloat32( 0.5 * float64ToFloat32( 1.0 - a ) );
+		x = sqrtf( z );
+		flag = 1;
+	}
+	else {
+		x = float64ToFloat32( a );
```

Comment:
```suggestion
		x = a;
```

You've already converted to float32 at L99.